cmd/maya-exporter/app/collector/zvol/pool_synctime.go

```diff
@@ -110,6 +114,10 @@ func (p *poolMetrics) get() *poolfields {
 // Collect is implementation of prometheus's prometheus.Collector interface
 func (p *poolMetrics) Collect(ch chan<- prometheus.Metric) {
 	p.Lock()
+	if p.isRequestInProgress() {
```

Comment:
Are we trying to prevent a deadlock from happening with this code?
Out of curiosity..
How does it get a lock when another request is in progress?

Reply:
It's only protecting request flag.

Reply:
@avishnu . Its not necessarily a deadlock. we are preventing requests from being queuing up.
We take the lock on the request flag, check if a request is in progress or not and release the lock. This will help us take lock only for a short time, and only for the single request flag, instead of the complete metric struct.